src/qibo/core/hamiltonians.py

```diff
@@ -252,14 +399,36 @@ def merge_one_qubit(self, terms):
             if target == pair[0]:
                 matrix = terms[pair]
             else:
-                c, m1, m2 = self.terms[pair]
+                matrices = self.terms[pair]
                 pair = (pair[1], pair[0])
-                matrix = c * K.np.kron(m2, m1)
+                matrix = 0
+                for i in range(0, len(matrices), 3):
+                    matrix += matrices[i] * self.multikron(matrices[i + 2: i:-1])
             eye = K.np.eye(2, dtype=matrix.dtype)
             merged[pair] = K.np.kron(one_qubit[target], eye) + matrix
         merged.update(two_qubit)
         return merged
 
+    def trotter_terms(self):
+        """Creates a dictionary of targets and matrices.
+
+        Useful for creating :class:`qibo.abstractions.hamiltonians.TrotterHamiltonian`
+        objects.
+
+        Returns:
+            terms (dict): Dictionary that maps tuples of targets to the matrix
+                          that acts on these on targets.
+            constant (float): The overall constant term of the Hamiltonian.
+        """
+        terms = {t: m for t, m in self.partial_matrices()}
+        if tuple() in terms:
+            constant = terms.pop(tuple()) + self.constant
+        else:
+            constant = self.constant
+        if set(len(t) for t in terms.keys()) == {1, 2}:
+            terms = self.merge_one_qubit(terms)
+        return terms, self.constant
```

Comment:
Thanks for the update, the issue is here:
```suggestion
        return terms, constant
```
Note that even after this fix there are some numerical issues with the Trotter of the MaxCut and we should set for example `atol=1e-12` to make this test pass. This is probably because the construction of the terms within the Trotter introduces a small numerical error and some values that should be 0 become `1e-16`. I don't think it is due to a bug.

Reply:
Thank you, this works.